File: Semestr 3/Paradygmaty_Programowania/Ocaml/lab_3.ml
let moduloList (list, m) = List.map (function x -> x mod m) list
	
let ( ^^ ) a b =
	match a,b with 
	| true, false 
	| false, true -> true
	| _ -> false

moduloList ([1;2;3;4;5;6;7;8], 2);;
moduloList ([1;3;5;7;9], 3);;

true ^^ true;;
true ^^ false;;
false ^^ true;;
false ^^ false;;

File: Semestr 3/Paradygmaty_Programowania/Scala/src/lab_3.scala
def potega (a:Float, n:Int):Float =
  if (n < 0) potega (1/a, -n)
  else n match{
    case 0 => 1
    case 1 => a
    case x => a*potega(a, n-1)
}

def potegaO (a:Float, n:Int):Float = {
  def pot (n:Int, acc:Float):Float =
    n match{
    case x if x < 0 => potegaO (1/a, -n)
    case 0 => 1
    case 1 => acc
    case x =>pot(n-1, a*acc)
  }
  pot(n, a)
}
  
  